navigation/index.tsx

```diff
@@ -55,8 +56,9 @@ const Stack = createNativeStackNavigator<RootStackParamList>();
 
 function RootNavigator() {
   const userContext = useContext(AppContext);
+  const authUserContext = useContext(AuthUserContext);
```

Comment:
nit, prefer de-structuring objects:
```suggestion
  const { user } = useContext(AuthUserContext);
``` 

and then just do `if(!user)` on 61.

For context, reasoning for this is it's usually better to see the shape of the object when you access it, rather than requiring reading through the rest of the block to know what's on the object